jablib/src/main/java/org/jabref/logic/util/io/RegExpBasedFileFinder.java

```diff
@@ -166,20 +165,22 @@ private List<Path> findFile(final BibEntry entry, final Path directory, final St
                 actualDirectory = Path.of(dirToProcess + '/');
                 continue;
             }
-            if (".".equals(dirToProcess)) { // Stay in current directory
-                continue;
-            }
-            if ("..".equals(dirToProcess)) {
-                actualDirectory = actualDirectory.getParent();
-                continue;
-            }
-            if ("*".equals(dirToProcess)) { // Do for all direct subdirs
-                File[] subDirs = actualDirectory.toFile().listFiles();
-                if (subDirs != null) {
-                    String restOfFileString = StringUtil.join(fileParts, "/", index + 1, fileParts.length);
-                    for (File subDir : subDirs) {
-                        if (subDir.isDirectory()) {
-                            resultFiles.addAll(findFile(entry, subDir.toPath(), restOfFileString, extensionRegExp));
+            switch (dirToProcess) {
+                case "." -> {
+                    continue;  // Stay in current directory
+                }
+                case ".." -> {
+                    actualDirectory = actualDirectory.getParent();
+                    continue;
+                }
+                case "*" -> {
+                    File[] subDirs = actualDirectory.toFile().listFiles();
+                    if (subDirs != null) {
+                        String restOfFileString = StringUtil.join(fileParts, "/", index + 1, fileParts.length);
+                        for (File subDir : subDirs) {
+                            if (subDir.isDirectory()) {
+                                resultFiles.addAll(findFile(entry, subDir.toPath(), restOfFileString, extensionRegExp));
+                            }
```

Comment:
In a follow-up, this could be modernized, too.

1. Rename to `currentDirectory` (`actual` is IMHO a false friend)
2. Try to use `walkFileTree` -> https://docs.oracle.com/javase/tutorial/essential/io/walk.html

Can't be done be Java newcomers as reviewing of the resulting code is too hard ^^